src/Shunting Yard/shuntingYard.c

```diff
@@ -0,0 +1,77 @@
+#include <stdio.h>
+#include <string.h>
+#include "list.h"
+#define MAX_INPUT_STRING_SIZE 1000
+
+int priority(char symbol);
+
+int main(void)
+{
+    Stack stack;
+    initializeStack(&stack);
+
+    char inputExpression[MAX_INPUT_STRING_SIZE];
+    printf("Введите выражение в инфиксной форме: ");
+    fgets(inputExpression, sizeof(inputExpression), stdin);
+    if (strcspn(inputExpression, "\n") != strlen(inputExpression)) {
+        inputExpression[strcspn(inputExpression, "\n")] = '\0';
+    }
+
+    unsigned int len = strlen(inputExpression);
+
+    printf("Выражение в постфиксной форме:");
+
+    for (unsigned int i = 0; i < len; i++) {
+        char currentSymbol = inputExpression[i];
+        if (currentSymbol != '(' && currentSymbol != ')' && currentSymbol != ' ') {
+            int currentSymbolPriority = priority(currentSymbol);
+            if (currentSymbolPriority == 3) {
+                printf(" %c", inputExpression[i]);
+            }
+            else if (currentSymbolPriority < 3 && !isStackEmpty(&stack)) {
+                char operate;
+                while (priority(operate = peek(&stack)) <= currentSymbolPriority) {
+                    printf(" %c", pop(&stack));
+                }
+                push(&stack, currentSymbol);
+            }
+            else if (currentSymbolPriority < 3 && isStackEmpty(&stack)) {
+                push(&stack, currentSymbol);
+            }
```

Comment:
Не надо повторяться:
```suggestion
            else {
                if (!isStackEmpty(&stack)) {
                    char operate;
                    while (priority(operate = peek(&stack)) <= currentSymbolPriority) {
                        printf(" %c", pop(&stack));
                    }
                }
                push(&stack, currentSymbol);
            }
```

Reply:
А ещё тут бывает underflow, если все операторы в стеке имеют меньший приоритет. Так что надо так:
```suggestion
            else {
                char operate;
                while (!isStackEmpty(&stack) && priority(operate = peek(&stack)) <= currentSymbolPriority) {
                    printf(" %c", pop(&stack));
                }
                push(&stack, currentSymbol);
            }
```